src/components/Gallery/Gallery.tsx

```diff
@@ -0,0 +1,152 @@
+import * as React from 'react';
+
+import {ArrowLeft, ArrowRight, Xmark} from '@gravity-ui/icons';
+import type {ModalProps} from '@gravity-ui/uikit';
+import {Button, Icon, Modal, Text, ThemeProvider, useThemeValue} from '@gravity-ui/uikit';
+
+import {block} from '../utils/cn';
+
+import type {GalleryItemProps} from './GalleryItem';
+import {FilesGalleryFallbackText} from './components/FallbackText';
+import type {UseNavigationProps} from './hooks/useNavigation';
+import {useNavigation} from './hooks/useNavigation';
+import {i18n} from './i18n';
+import {getInvertedTheme} from './utils/getInvertedTheme';
+
+import './Gallery.scss';
+
+const cnGallery = block('gallery');
+
+const emptyItems: GalleryItemProps[] = [];
+
+export type GalleryProps = {
+    fullScreen?: boolean;
+    modalClassName?: string;
+    className?: string;
+    children?: React.ReactElement<GalleryItemProps>[];
+    invertTheme?: boolean;
+    noItemsMessage?: string;
+} & Pick<ModalProps, 'open' | 'container'> &
+    Required<Pick<ModalProps, 'onOpenChange'>> &
+    Pick<UseNavigationProps, 'initialItemIndex'>;
+
+export const Gallery = ({
+    initialItemIndex,
+    open,
+    onOpenChange,
+    fullScreen,
+    container,
+    modalClassName,
+    className,
+    invertTheme,
+    children,
+    noItemsMessage,
+}: GalleryProps) => {
+    const items = children ? React.Children.map(children, (child) => child.props) : emptyItems;
+    const theme = useThemeValue();
+
+    const {activeItemIndex, setActiveItemIndex, handleGoToNext, handleGoToPrevious} = useNavigation(
+        {
+            itemsCount: items.length,
+            initialItemIndex,
+            selectedPreviewItemClass: `.${cnGallery('preview-list-item')}_selected`,
+        },
+    );
+
+    const handleClose = React.useCallback(() => {
+        onOpenChange?.(false);
+    }, [onOpenChange]);
+
+    const activeItem = items[activeItemIndex] || items[0];
+    console.log(activeItem);
+    return (
+        <Modal
+            container={container}
+            className={cnGallery({mode: fullScreen ? 'full-screen' : 'default'}, modalClassName)}
+            open={open}
+            onOpenChange={onOpenChange}
+            disableEscapeKeyDown={fullScreen}
+        >
+            <ThemeProvider theme={invertTheme ? getInvertedTheme(theme) : theme}>
+                <div className={cnGallery('content', className)}>
```

Comment:
`className` prop is expected to be on the root node. This is confusing. Why we need to classes, for modal and for content?